### PR TITLE
Removed core dependency on raylib, migrated game example render and input to internal modules

### PR DESCRIPTION
**Pull request by J-Mo63**
### Description

This (much awaited) PR aims to entirely remove the raylib dependency from the core module (closing #52), replacing its input and render functionality used in the game example, with the window and render modules respectively.

Build scripts have been updated accordingly and many core systems have been repatriated in the process to accomodate the migration. As such, please read through the changes carefully to avoid any missing behaviour being lost.

### The PR has been...
- [x] provided a reasonable name that is not just the branch name (e.g "Added Vulkan render delegate")
- [x] linked to its related issue
- [x] assigned a reviewer from the team
- [x] labelled appropriately

### The code has been...
- [x] made mergable and free of conflicts in relation to `master` *(according to GitHub)*
- [x] tested in a packaged state using the `package` targets
- [x] pulled to the reviewer's machine and reasonably tested

### CODE REVIEW

File: examples/render/src/main.cpp [hunk 1]
```diff
@@ -147,14 +146,14 @@ int main()
         currentTime = newTime;
 
         window.Update();
-        window.ToggleCursor(inputEnabled);
```
Comment (J-Mo63) [inline examples/render/src/main.cpp:149]: Re-add grid display call for 3D

File: examples/game/src/entities/FPSCamera.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
+//
+//
```
Comment (J-Mo63) [inline examples/game/src/entities/FPSCamera.cpp:2]:
```suggestion
```

File: examples/game/src/entities/Player.cpp [hunk 3]
```diff
@@ -9,38 +9,37 @@
...
 // Static member initialisation
 const Siege::String Player::ENTITY_NAME("Player");
 
 void Player::OnStart()
 {
     // Register the entity with systems
-    Siege::String modelPath = "models/cube/cube.obj";
-    Siege::String texturePath = "models/cube/cube.png";
-    Siege::Statics::Render().Add(this, {modelPath, texturePath});
 }
```
Comment (J-Mo63) [inline examples/game/src/entities/Player.cpp:24]: Whole function can go

File: examples/game/src/entities/Player.cpp [hunk 4]
```diff
@@ -53,7 +52,6 @@ void Player::OnUpdate()
 void Player::OnDestroy()
 {
     // Deregister the entity from systems before freeing it
-    Siege::Statics::Render().Remove(this);
 }
```
Comment (J-Mo63) [inline examples/game/src/entities/Player.cpp:55]: This can go for now

**Review by J-Mo63 — COMMENTED** [review]
Looks good, let's finalise these last few things, fix Windows and merge.

File: engine/core/Makefile [hunk 5]
```diff
@@ -20,31 +20,16 @@ ifeq ($(platform), windows)
...
-raylibIncludeDir := $(vendorDir)/include/raylib
-raylibLib := $(vendorDir)/raylib-cpp/vendor/raylib/$(libGenDir)/libraylib.a
 
 # Build the static library
-$(coreLib): $(raylibLib) $(raylibIncludeDir) $(coreObjects)
+$(coreLib): $(coreObjects)
 	$(MKDIR) $(call platformpth, $(libDir))
-	$(RM) $(call platformpth, $(coreLib))
-	$(call COMBINE_LIBS, $(vendorDir)/raylib-cpp/vendor/raylib/$(libGenDir)/libraylib.a, $(coreObjects), $(libDir), core)
+	ar -crs $(coreLib) $(coreObjects)
```
[the remark this answers is not in the document]
Comment (J-Mo63) [reply engine/core/Makefile:27]: The combinelibs scripts will need some attention to support scenarios where no libs are being combined.

**Review by Raelr — APPROVED** [review]

**Pull request merged**